dnadoh/cache.py

```diff
@@ -0,0 +1,118 @@
+"""Simulate a local cache for remote files.
+
+Tools like DataCorral download files from the cloud on demand.
+If there's already a local copy of the file, DC re-uses that
+rather than downloading the file again.
+
+`FileCache` simulates this by "downloading" (copying) files
+from one local directory to another.  When a caller asks for
+a file called "tammy.csv", `FileCache` downloads the file if
+its name isn't already in the cache, then returns the path
+to the local copy of the file.
+
+Things to change:
+
+1.  Download from URLs using the `requests` library.
+2.  Make `FileCache` configurable so that a caller can specify
+    a base URL for remote files and the local cache directory
+    (but only has to specify them once). - done
+3.  Make `FileCache` able to download both remote files and
+    local files.
+4.  Check that files whose names are in the cache are actually
```

Comment:
check that if you have the file name in the cache dictionary, that it's actually on disk (not that it's actually the same file)